nixos/modules/services/x11/redshift.nix

```
@@ -88,51 +71,101 @@ in {
       example = "/bin/redshift-gtk";
       description = ''
         Redshift executable to use within the package.
+
+        To use Gammastep, pass either <literal>/bin/gammastep</literal>
+        or <literal>/bin/gammastep-indicator</literal>.
       '';
     };
 
-    extraOptions = mkOption {
-      type = types.listOf types.str;
-      default = [];
-      example = [ "-v" "-m randr" ];
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+      };
```

```suggestion
      type = types.submodule ({ options, ... }: {
        freeformType = settingsFormat.type;

        options.redshift = mkOption {
          type = settingsFormat.type.functor.wrapped;
        };

        # Copy all redshift definitions to general, such that in case of gammastep,
        # which prefers general over redshift, gets both values from general and redshift
        # While the original redshift just gets values from redshift, as it should
        config.general = lib.modules.mkAliasAndWrapDefsWithPriority id options.redshift;
      });
```

This then obviates the need for the `mainSection` hack below. You can then just use `redshift` for the definitions to be picked up by both redshift and gammastep.

---

I don't know if copying would work. `gammastep` actually supports both `redshift` and `general` sections (but emits a warning in case `redshift` section is found).

I can tests if this works, but still looks wrong.

---

Also, I am quite sure that `redshift` actually checks for invalid inputs and gives an error in case of them being set. But let me try first and I will post the results later on.

---

Oh, yeah my assumption was that redshift ignores the `general` section, and gammastep ignores the `redshift` section if `general` exists. Knowing the behavior exactly would be great to find a good solution here.

---

Well, this actually works TIL. Ok, I think it is fair, going to do like you said here.